server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -1451,6 +1459,11 @@ public void run() {
                             return;
                         }
 
+                        if (localNodeCommissioned == false) {
```

Comment:
Every time the OpenSearch process is bounced, ```localNodeCommissioned```  will get reset to true. Now suppose, there is a decommissioned master-eligible node in the cluster and also at that point there is no active leader in the cluster (temporary, no quorum loss). Then this node can become candidate and will collect votes from others and become master. It may end up abdicating later ( we need to cross check that flow again) but it would cause un-necessary churn in the cluster. To mitigate this, JoinExecutor should also ensure node which is going to become active leader is also not decommissioned.

Reply:
Okay

Reply:
Can we have a stale lastAccepted State or none for a newly joining master-eligible node

Reply:
`lastAcceptedState` is the state last written on its disk. If its last state thinks it was decommissioned we can let it stay out of pre voting and let other master eligible node form the cluster. In its ping request, it would be able to join the cluster if it was not decommissioned

Reply:
Also, `lastAcceptedState` is also considered as source of truth to decide whether it can win election or not (based on configuration and exclusion)